### PR TITLE
grafanaPlugins.grafana-oncall-app: init at 1.5.1

### PR DESCRIPTION
**Pull request by greaka**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by loispostula — APPROVED** [review]

File: pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ grafanaPlugin, lib }:
+
+grafanaPlugin rec {
+  pname = "grafana-oncall-app";
+  version = "1.3.106";
```
Comment (emilylange) [inline pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix:5]: v1.5.0 has been released a week ago

https://github.com/grafana/oncall/releases/tag/v1.5.0

File: pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  zipHash = {
+    x86_64-linux = "sha256-8TL8HqiYCUxY2tCItpavYb77rGIpcJBWexwykFNWONg=";
+    aarch64-linux = "sha256-4rCj+NaKPZbuVohlKmSf1M6m5ng9HZMrwzBCgLPdiok=";
+    x86_64-darwin = "sha256-bpey6EwwAqXgxjvjJ6ou4rimidHCpUr+Z89YpAZK7z8=";
+    aarch64-darwin = "sha256-u/U2lu4szf9JFt/zfhGmWKF2OUqpJDNaSI69EDdi1+w=";
+  };
+  meta = with lib; {
+    description =
+      "Collect and analyze alerts, escalate based on schedules and deliver them to Slack, Phone Calls, SMS and others.";
```
Comment (emilylange) [inline pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix:14]: > `meta.description` must:
> - Be short, just one sentence.
> - [...]
> - Not end with a period (or any punctuation for that matter).

https://github.com/NixOS/nixpkgs/tree/c10fb0df312867b0ae47a580818b317a9ae61f21/pkgs#meta-attributes

How about (taking the beginning of the official GitHub repo description):

```suggestion
    description = "Developer-friendly incident response for Grafana";
```

**Review by emilylange — CHANGES_REQUESTED** [review]
Commit should be titled
`grafanaPlugins.grafana-oncall-app: init at 1.3.106` instead of
`grafana-oncall-app: init at 1.3.106`

File: pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{ grafanaPlugin, lib }:
+
+grafanaPlugin rec {
+  pname = "grafana-oncall-app";
+  version = "1.5.0";
+  zipHash = {
+    x86_64-linux = "sha256-DOQ3W77fhLedgXH1VHW/JTjnDi0PxeASdsSAsnyG2xg=";
+  };
```
Comment (emilylange) [inline pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix:8]: Your previous version had hashes for all 4 architectures, not just `x86_64-linux`.

But given this plugin does not contain precompiled (architecture-specific) binaries and

https://grafana.com/api/plugins/grafana-oncall-app/versions/1.5.0/download?os=linux&arch=amd64
and
https://grafana.com/api/plugins/grafana-oncall-app/versions/1.5.0/download

return exactly the same ZIP, file structure, contents and recursive NAR hash, this should probably use

```suggestion
  zipHash = "sha256-DOQ3W77fhLedgXH1VHW/JTjnDi0PxeASdsSAsnyG2xg=";
```
instead.

File: pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  meta = with lib; {
+    description = "Developer-friendly incident response for Grafana";
+    license = licenses.agpl3Only;
+    maintainers = lib.teams.fslabs.members;
+    platforms = attrNames zipHash;
```
Comment (emilylange) [inline pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix:13]:
```suggestion
    platforms = platforms.unix;
```

**Review by emilylange — CHANGES_REQUESTED** [review]

**Comment by greaka** [comment]
Thanks for all the feedback :slightly_smiling_face:

**Comment by emilylange** [comment]
@ofborg build grafanaPlugins.grafana-oncall-app

File: pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{ grafanaPlugin, lib }:
+
+grafanaPlugin {
+  pname = "grafana-oncall-app";
+  version = "1.5.1";
+  zipHash = "sha256-3mC4TJ9ACM9e3e6UKI5vaDwXuW6RjbsOQFJU5v0wjk8=";
+  meta = with lib; {
+    description = "Developer-friendly incident response for Grafana";
+    license = licenses.agpl3Only;
+    maintainers = lib.teams.fslabs.members;
```
Comment (emilylange) [inline pkgs/servers/monitoring/grafana/plugins/grafana-oncall-app/default.nix:10]:
```suggestion
    maintainers = teams.fslabs.members;
```

`lib` is already in the scope due to `with lib;` in line 7.

Probably not worth going through another review cycle, but something I noticed only now.

**Review by emilylange — APPROVED** [review]

**Comment by emilylange** [comment]
Both of you will receive automatic invites to the GitHub org soon.

This PR can be backported to `24.05`, if desired.

If you want, you can add the "backport release-24.05" label yourself after you've accepted the invite.

Either way, welcome to nixpkgs :)

**Pull request merged**